controllers/storagecluster/cephfilesystem.go

```diff
@@ -121,11 +122,86 @@ func (obj *ocsCephFilesystems) ensureCreated(r *StorageClusterReconciler, instan
 				return reconcile.Result{}, err
 			}
 		}
+		// create default csi subvolumegroup for the filesystem
+		// skip for the ocs provider mode
+		if !instance.Spec.AllowRemoteStorageConsumers {
+			err = r.createDefaultSubvolumeGroup(cephFilesystem.Name, cephFilesystem.Namespace, cephFilesystem.ObjectMeta.OwnerReferences)
+			if err != nil {
+				return reconcile.Result{}, err
+			}
+		}
 	}
 
 	return reconcile.Result{}, nil
 }
 
+func (r *StorageClusterReconciler) createDefaultSubvolumeGroup(filesystemName, filesystemNamespace string, ownerReferences []metav1.OwnerReference) error {
+	// TODO: After fix of rook issue https://github.com/rook/rook/issues/13220 add svg name spec
+
+	existingsvg := &cephv1.CephFilesystemSubVolumeGroup{}
+	err := r.Client.Get(r.ctx, types.NamespacedName{Name: "csi", Namespace: filesystemNamespace}, existingsvg)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.Log.Info("Uninstall: csi subvolumegroup not found.", "Subvolumegroup", klog.KRef(filesystemNamespace, "csi"))
+			return nil
+		}
+		r.Log.Error(err, "Uninstall: Unable to retrieve subvolumegroup.", "subvolumegroup", klog.KRef(filesystemNamespace, "csi"))
+		return fmt.Errorf("uninstall: Unable to retrieve csi subvolumegroup : %v", err)
```

Comment:
We shouldn't log as `Uninstall: ...`, we are in the creation loop, if the CR doesn't exist then we simply log `Creating ...` and create the CR in the next line.
Check the above code of `cephFileSystem` creation for ref.

Reply:
Updated, thanks for the feedback